components/formats-bsd/src/loci/formats/in/OBFReader.java

```diff
@@ -494,7 +553,38 @@ else if (isFLIMLabel(label)) {
           stepLabels.add(list);
         }
         meta_data.seriesMetadata.put("StepLabels", stepLabels);
+
+        in.skipBytes(obsoleteMetadataLength);
+
+        if (numFlushPoints > 0) {
+            List<Long> flushPoints = new ArrayList<>((int)numFlushPoints);
```

Comment:
Can you explain why the cast to ```int``` here is safe, or rework to handle the case when ```numFlushPoints``` is greater than ```Integer.MAX_VALUE```?

Reply:
For now I think it is relatively safe (though I've added an exception to detect this specific case) as we are hopefully nowhere near `Integer.MAX_VALUE` flush points (or chunks) when generating datasets yet. This would mean that the file contains > ~16GB of flush point data (per stack) or even ~32GB of chunk information (per stack). If we ever get near that we'll need an entirely different approach to processing and storing this information.